### PR TITLE
Fix log sites

### PR DESCRIPTION
**Pull request by erwinmombay**
This parses the message argument pulling out references (replacing them with %s) and template strings and spreads them as arguments if the method is variadic.

### CODE REVIEW

File: build-system/eslint-rules/no-non-string-log-args.js [hunk 1]
```diff
@@ -90,7 +95,7 @@ module.exports = function(context) {
       }
 
       const methodInvokedName = callee.property.name;
-      // Find the position of the argument we care about.
+      // Find the position of the argumen we care about.
```
Comment (kristoferbaxter) [inline build-system/eslint-rules/no-non-string-log-args.js]: Nit: argument

File: build-system/eslint-rules/no-non-string-log-args.js [hunk 2]
```diff
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+
```
Comment (kristoferbaxter) [inline build-system/eslint-rules/no-non-string-log-args.js]: Nit: Remove this extra line.

**Comment by erwinmombay** [comment]
got 1 major bug where I might be clobbering any var args already being passed in. need to preserve arguments that are already in place

**Comment by jridgewell** [comment]
I forked astexplorer to add autofix, follow https://github.com/jridgewell/astexplorer/tree/autofix#build-your-own-version-for-development to run locally.

This appears to not handle literals (like numbers) as the things that are concatenated to the string. It also seems to infinite loop with template literal concats.

**Comment by erwinmombay** [comment]
@jridgewell PTAL. replaced my code with your implementation. only minor change i did was to escape the single quotes before it replaces the node. appreciate it!

File: build-system/eslint-rules/no-non-string-log-args.js [hunk 3]
```diff
@@ -66,62 +78,130 @@ function getMetadata(name) {
...
+                  arg = source.getText(node.arguments[argI]);
+                  argI++;
+                } else {
+                  arg = sourceText;
+                }
+                args.push(arg);
+                return '%s';
+              });
+
+              message = message.replace(/'/g, '\\\'');
```
Comment (erwinmombay) [inline build-system/eslint-rules/no-non-string-log-args.js]: @jridgewell only change i did

**Pull request merged**